### PR TITLE
Feature add hackathon

### PR DESCRIPTION
**Pull request by tjclawson**
# Description

Add ability to post hackathon to api through app

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [X] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: app/src/main/java/com/lambdaschool/hackathon_portal/HackathonRepository.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    fun postHackathon(hackathon: Hackathon): LiveData<String> {
+        val bearerToken = "Bearer ${CurrentUser.currentUser.accessToken}"
+        hackathonService.postHackathon(CurrentUser.currentUser.id?.toInt(), bearerToken, hackathon)
+            .enqueue(object: Callback<Hackathon> {
+                override fun onFailure(call: Call<Hackathon>, t: Throwable) {
+                    addHackathonResponse.value = "Failed to connect to API"
+                    Log.i(REPO_TAG, "Failed to connect to API")
+                    Log.i(REPO_TAG, t.message.toString())
```
Comment (05nelsonm) [inline app/src/main/java/com/lambdaschool/hackathon_portal/HackathonRepository.kt]: Display error to user via Toast?

File: app/src/main/java/com/lambdaschool/hackathon_portal/di/FragmentComponent.kt [hunk 2]
```diff
@@ -26,4 +27,5 @@ interface FragmentComponent {
     fun injectDashboardFragment(fragment: DashboardFragment)
     fun injectAccountFragment(fragment: AccountFragment)
     fun injectSettingsFragment(fragment: SettingsFragment)
+    fun injectAddHackathonFragment(fragmen: AddHackathonFragment)
```
Comment (05nelsonm) [inline app/src/main/java/com/lambdaschool/hackathon_portal/di/FragmentComponent.kt]: typo in `fragment`

Change to `CreateHackathonFragment` to be more descriptive?

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/MainActivity.kt [hunk 3]
```diff
@@ -77,6 +77,10 @@ class MainActivity : AppCompatActivity() {
                     nav_host_fragment.findNavController().navigate(R.id.settingsFragment)
                     title = "Settings"
                 }
+
+                R.id.nav_drawer_add_hackathon -> {
+                    nav_host_fragment.findNavController().navigate(R.id.addHackathonFragment)
```
Comment (05nelsonm) [inline app/src/main/java/com/lambdaschool/hackathon_portal/ui/MainActivity.kt]: My oversight on PR #12. Forgot to refactor these calls.

navController is injected on line 40 and can be called directly

```suggestion
                    navController.navigate(R.id.addHackathonFragment)
```

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 4]
```diff
@@ -17,6 +17,10 @@
             android:id="@+id/nav_drawer_logout"
             android:title="@string/logout" />
 
+        <item
```
Comment (05nelsonm) [inline app/src/main/res/menu/activity_main_drawer.xml]: Move above `settings` & `logout` items?

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/add/AddHackathonFragment.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        switch_is_open.setOnCheckedChangeListener { button, b ->
+            switchState = when (b) {
+                true -> true
+                false -> false
+            }
+        }
+
+        fab_save_hackathon.setOnClickListener {
+            val newHackathon = Hackathon(edit_text_hackathon_name.text.toString(),
```
Comment (05nelsonm) [inline app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/add/AddHackathonFragment.kt]: Add logic to to check if `name`, `start_date`, & `end_date` are empty before submitting a network call.

File: app/src/main/java/com/lambdaschool/hackathon_portal/HackathonRepository.kt (hunk 1 above)
Comment (tjclawson) [reply app/src/main/java/com/lambdaschool/hackathon_portal/HackathonRepository.kt]: Implemented in commit 4f768f5f09460f6d65b371a974ff92469d63b2a7

File: app/src/main/java/com/lambdaschool/hackathon_portal/di/FragmentComponent.kt (hunk 2 above)
Comment (tjclawson) [reply app/src/main/java/com/lambdaschool/hackathon_portal/di/FragmentComponent.kt]: Fixed in commit f4fe16d5d80bccd8f11760acbf005fe5a10b7359

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/MainActivity.kt (hunk 3 above)
Comment (tjclawson) [reply app/src/main/java/com/lambdaschool/hackathon_portal/ui/MainActivity.kt]: Changed in commit 3b1138a1fc3b243d79ce1c20727eac01c3287d03

File: app/src/main/res/menu/activity_main_drawer.xml (hunk 4 above)
Comment (tjclawson) [reply app/src/main/res/menu/activity_main_drawer.xml]: Moved and refactored name to Create Hackathon in commit d32666441ae969c5e7a7ef9d87d3396db1105624

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/add/AddHackathonFragment.kt (hunk 5 above)
Comment (tjclawson) [reply app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/add/AddHackathonFragment.kt]: Implemented in commit a646269149804cc3f463b9271fb2cd0413b974dd

**Review by 05nelsonm — APPROVED** [review]

**Pull request merged**